### PR TITLE
Fix single image bug and tag dicoms for reaper

### PR DESCRIPTION
**Pull request by hkethi002**
Fixes bug where only one image file was being sent per dicom. 
Sets tag for dicom files for reaper to ignore

### CODE REVIEW

File: tagger.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+def tag_folder(dir_path, group, identifier, tag_value):
+    for image_file in os.listdir(dir_path):
+        if '.dcm' in image_file:
+            tag_image(os.path.join(dir_path, image_file), group, identifier, tag_value)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='dicom-send tag')
+    parser.add_argument("dicom-input")
```
Comment (ambrussimon) [inline tagger.py]: Positional args have a bug where [dashes are not translated to underscores](https://bugs.python.org/issue15125) thus `args.dicom_input` below will fail. (I suggest just renaming to `dicom`)

File: tagger.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    parser.add_argument('--group', '-g', type=str, required=True, help='The group of the private element tag')
+    parser.add_argument('--identifier', '-i', type=str, required=True, help='Identifier')
+    parser.add_argument('--tag-value', '-t', type=str, required=True, help='String to tag the dicom files')
+
+    args = parser.parse_args()
+
+    if os.path.isdir(args.dicom_input):
+        tag_folder(args.dicom_input, args.group, args.identifier, args.tag_value)
```
Comment (ambrussimon) [inline tagger.py]: At this point `args.group == "0x0021"`, and I was almost getting excited about `pydicom` handling even that. Turns out it doesn't, please cast `args.group = int(args.group, 16)` just after `parse_args()`.

File: tagger.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import os
+import pydicom
+import argparse
+
+
+class TagError(Exception):
+    pass
+
+
+def add_send_tag(dicomfile, group, identifier, tag_value):
```
Comment (ambrussimon) [inline tagger.py]: I like how `tagger.py` is generic; the only minor outlier is `send_tag` in this function name.

File: tagger.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                        return
+                    elif private_elem.value.lower().startswith(tag_value.lower()):
+                        return
+        else:
+            identifier_tag = (group, elem_tag)
+            dicomfile.add_new(identifier_tag, 'LO', identifier)
+            private_tag = (group, elem_tag * 0x0100)
+            dicomfile.add_new(private_tag, 'LO', tag_value)
+            return
+    raise TagError('No free element in group {} to tag the dicom'.format(group))
```
Comment (ambrussimon) [inline tagger.py]: Without calling `dicomfile.save_as(file_path)` the changes (if any) aren't committed to the file on disk.

File: tagger.py (hunk 2 above)
Comment (hkethi002) [reply tagger.py]: I believe at least as far back as pydicom v1.0.1, I get the following:
```
>>> from pydicom import tag
>>> t1 = tag.Tag("0x0021", "0x0010")
>>> t2 = tag.Tag(0x0021, 0x0010)
>>> t1 ==t2
True
```

File: tagger.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+class TagError(Exception):
+    pass
+
+
+def add_private_tag(dicomfile, group, identifier, tag_value):
+    for elem_tag in range(0x0010, 0x00FF):
+        dataelem = dicomfile.get((group, elem_tag))
+        if dataelem:
+            is_private_creator = dataelem.tag.group == group and 0x0010 <= dataelem.tag.element <= 0x00ff
```
Comment (ehlertjd) [inline tagger.py]: Under what circumstances is this `False`? Does pydicom return a different element than you asked for?

File: tagger.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def tag_folder(dir_path, group, identifier, tag_value):
+    for image_file in os.listdir(dir_path):
+        if '.dcm' in image_file:
+            tag_image(os.path.join(dir_path, image_file), group, identifier, tag_value)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='dicom-send tag')
```
Comment (ehlertjd) [inline tagger.py]: I think this script should log what it does. (i.e. what offsets it added)

File: run [hunk 7]
```diff
@@ -92,6 +95,10 @@ for input_file in $INPUT_DIR/file/*; do
     cp -a $input_file $INPUT_DIR
   fi
 
+  ##############################################################################
+  # Tag all of the files so that they are not re-reaped
+  python tagger.py "$dicom_input" -g '0x0021' -i 'Flywheel' -t 'DICOM Send'
```
Comment (ehlertjd) [inline run]: Highly recommend that you check your exit code here, and do **NOT** proceed with sending if tagging fails.

File: tagger.py (hunk 5 above)
Comment (hkethi002) [reply tagger.py]: It's a check to see if that tag is free, if there is no element there it returns None

Comment (ehlertjd) [reply tagger.py]: Question is about Line 14: `is_private_creator`

Comment (hkethi002) [reply tagger.py]: Ah, yeah that should be removed

File: tagger.py [hunk 8]
```diff
@@ -52,6 +54,11 @@ def tag_folder(dir_path, group, identifier, tag_value):
...
 
     if os.path.isdir(args.dicom):
-        tag_folder(args.dicom, args.group, args.identifier, args.tag_value)
+        tags = tag_folder(args.dicom, args.group, args.identifier, args.tag_value)
     else:
-        tag_image(args.dicom, args.group, args.identifier, args.tag_value)
+        tags = [tag_image(args.dicom, args.group, args.identifier, args.tag_value)]
+
+    if len(tags) > 1:
+        print('WARNING: Tagged at different elements')
```
Comment (ehlertjd) [inline tagger.py]: 👌

**Comment by gsfr** [comment]
Late to the party. Thanks everyone for cranking this out.

**Pull request merged**